src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexWriteOperation.cs

```diff
@@ -60,14 +68,48 @@ public class IndexWriteOperation : IndexOperationBase
             {
                 throw new IndexWriteException(e);
             }
+
+            var mapReduceIndex = index as MapReduceIndex;
```

Comment:
Maybe create new implementation for IndexWriteOperation and overwrite needed parts?